src/Orbit/main.cpp

```diff
@@ -283,27 +283,37 @@ static void TryMoveSavedDataLocationIfNeeded() {
     return;
   }
 
+  std::thread::id main_thread_id = std::this_thread::get_id();
+
   orbit_qt::MoveFilesDialog dialog;
   orbit_qt::MoveFilesProcess process;
 
-  QObject::connect(&process, &orbit_qt::MoveFilesProcess::generalError,
-                   [&dialog](const std::string& error_message) {
+  QObject::connect(&process, &orbit_qt::MoveFilesProcess::generalError, &dialog,
+                   [&dialog, main_thread_id](const std::string& error_message) {
+                     CHECK(main_thread_id == std::this_thread::get_id());
```

Comment:
Why this check? The behaviour of `QObject::connect` is guaranteed by Qt. When could this fail?

Reply:
Ah, I just read the description which mentions the CHECK. I personally find it confusing because I immediately think about when this could fail and I think it can't. But maybe that's just me. Feel free to ignore.

Reply:
I know about the guarantee. This cannot fail currently.@dimitry- suggested to leave the check in. I guess for clarity while reading the code. I do not feel strongly about this. @dimitry- maybe you have a opinion on this.

Reply:
without the pointer to object (dialog in this case) there is no guarantee, which was not obvious to me, I assumed it will take the caller thread, hence I find the check useful but again this is my lack of knowledge of qt things.

Reply:
I see. That makes sense to me. Thanks for the explanation!